File: gameserver/src/main/java/l2s/gameserver/network/authcomm/as2gs/KickPlayer.java
package l2s.gameserver.network.authcomm.as2gs;

import io.netty.buffer.ByteBuf;
import l2s.gameserver.model.Player;
import l2s.gameserver.network.authcomm.AuthServerClientService;
import l2s.gameserver.network.authcomm.ReceivablePacket;
import l2s.gameserver.network.authcomm.vertx.AuthServerConnection;
import l2s.gameserver.network.l2.GameClient;
import l2s.gameserver.network.l2.components.SystemMsg;
import l2s.gameserver.network.l2.s2c.ServerCloseSocketPacket;

public class KickPlayer extends ReceivablePacket {
    private final String account;

    public KickPlayer(AuthServerConnection client, ByteBuf byteBuf) {
        super(client, byteBuf);

        account = readString(byteBuf);
    }

    @Override
    protected void runImpl(AuthServerConnection client) {
        GameClient gameClient = AuthServerClientService.INSTANCE.removeWaitingClient(account);
        if (gameClient == null)
            gameClient = AuthServerClientService.INSTANCE.removeAuthedClient(account);
        if (gameClient == null)
            return;
        Player activeChar = gameClient.getActiveChar();
        if (activeChar != null) {
            activeChar.sendPacket(SystemMsg.ANOTHER_PERSON_HAS_LOGGED_IN_WITH_THE_SAME_ACCOUNT);
            activeChar.kick();
        } else
            gameClient.close(ServerCloseSocketPacket.STATIC);
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/as2gs/KickPlayer.kt
package l2s.gameserver.network.authcomm.as2gs

import io.netty.buffer.ByteBuf
import l2s.gameserver.network.authcomm.AuthServerClientService
import l2s.gameserver.network.authcomm.ReceivablePacket
import l2s.gameserver.network.authcomm.vertx.AuthServerConnection
import l2s.gameserver.network.l2.GameClient
import l2s.gameserver.network.l2.components.SystemMsg
import l2s.gameserver.network.l2.s2c.ServerCloseSocketPacket

class KickPlayer(client: AuthServerConnection, byteBuf: ByteBuf) : ReceivablePacket(client, byteBuf) {

    private val account: String = readString(byteBuf)

    override fun runImpl(client: AuthServerConnection) {
        var client: GameClient? =
            AuthServerClientService.removeWaitingClient(account)
        if (client == null) client = AuthServerClientService.removeAuthedClient(account)
        if (client == null) return
        val activeChar = client.activeChar
        if (activeChar != null) {
            //FIXME [G1ta0] сообщение чаще всего не показывается, т.к. при закрытии соединения очередь на отправку очищается
            activeChar.sendPacket(SystemMsg.ANOTHER_PERSON_HAS_LOGGED_IN_WITH_THE_SAME_ACCOUNT)
            activeChar.kick()
        } else {
            client.close(ServerCloseSocketPacket.STATIC)
        }
    }
}